apps/AEPSampleAppNewArchEnabled/app/TargetView.tsx

```diff
@@ -0,0 +1,213 @@
+import {
+  Target,
+  TargetOrder,
+  TargetParameters,
+  TargetPrefetchObject,
+  TargetProduct,
+  TargetRequestObject,
+} from '@adobe/react-native-aeptarget';
+import {  useRouter } from 'expo-router';
+
+function TargetView() {
+  const targetExtensionVersion = async () => {
+    const version = await Target.extensionVersion();
+    console.log(`AdobeExperienceSDK: Target version: ${version}`);
+  };
+
+  const clearPrefetchCache = () => Target.clearPrefetchCache();
+
+  const getSessionId = async () => {
+    const sessionId = await Target.getSessionId();
+    console.log(`Session ID: ${sessionId}`);
+  };
+
+  const getThirdPartyId = async () => {
+    const id = await Target.getThirdPartyId();
+    console.log(`AdobeExperienceSDK: Third Party ID: ${id}`);
+  };
+
+  const getTntId = async () => {
+    const id = await Target.getTntId();
+    console.log(`AdobeExperienceSDK: TNT ID ${id}`);
+  };
+
+  const resetExperience = () => Target.resetExperience();
+
+  const setPreviewRestartDeeplink = () =>
+    Target.setPreviewRestartDeeplink('https://www.adobe.com');
+
+  const setSessionId = () => Target.setSessionId('sessionId');
+
+  const setTntId = () => Target.setTntId('tntId');
+
+  const setThirdPartyId = () => Target.setThirdPartyId('thirdPartyId');
+
+  const retrieveLocationContent = () => {
+    const mboxParameters1 = {status: 'platinum'};
+    const mboxParameters2 = {userType: 'Paid'};
+    const purchaseIDs = ['34', '125'];
+
+    const targetOrder = new TargetOrder('ADCKKIM', 344.3, purchaseIDs);
+    const targetProduct = new TargetProduct('24D3412', 'Books');
+    const parameters1 = new TargetParameters(mboxParameters1);
+    const request1 = new TargetRequestObject(
+      'sdk_smoke_tests_target',
+      parameters1,
+      'defaultContent1',
+      (error, content) => {
+        if (error) {
+          console.error(error);
+        } else {
+          console.log('Adobe content:' + content);
+        }
+      },
+    );
+
+    const parameters2 = new TargetParameters(
+      mboxParameters2,
+      {profileParameters: 'parameterValue'},
+      targetProduct,
+      targetOrder,
+    );
+    const request2 = new TargetRequestObject(
+      'aep-loc-2',
+      parameters2,
+      'defaultContent2',
+      (error, content) => {
+        if (error) {
+          console.error(error);
+        } else {
+          console.log('Adobe content:' + content);
+        }
+      },
+    );
+
+    const locationRequests = [request1, request2];
+    const profileParameters1 = {ageGroup: '20-32'};
+
+    const parameters = new TargetParameters(
+      {parameters: 'parametervalue'},
+      profileParameters1,
+      targetProduct,
+      targetOrder,
+    );
+    Target.retrieveLocationContent(locationRequests, parameters);
+  };
+
+  const displayedLocations = () =>
+    Target.displayedLocations(['sdk_smoke_tests_target', 'aep-loc-2', 'sdk_smoke_tests_target_a4t']);
```

Comment:
You can pass in TargetParameter here if we want to maintain parity across platforms

Reply:
code updated